accelerometer/summariseEpoch.py

```diff
@@ -406,8 +417,123 @@ def calculateECDF(e, inputCol, summary):
         summary[inputCol + '-ecdf-' + str(accUtils.formatNum(x,0)) + 'mg'] = \
             accUtils.formatNum(ecdf, 5)
 
+def calculatePSD(e, epochPeriod, fourierWithAcc, labels, summary):
+    """Calculate the power spectral density from fourier analysis of a 1 day frequency
+    
+    :param pandas.DataFrame e: Pandas dataframe of epoch data
+    :param int epochPeriod: Size of epoch time window (in seconds)
+    :param bool fourierWithAcc:True calculates fourier done with acceleration data instead of sleep data
+    :param list(str) labels: Activity state labels
+    :param dict summary: Output dictionary containing all summary metrics
+
+    :return: Write dict <summary> keys 'PSD-<W/Hz>'
+    """
+    if fourierWithAcc:
+        y = (e['accImputed'])
+    else:
+        cols = []
+        # get imputed variable names for each activity type
+        # NB - make sure sleep comes first
+        for accType in labels:
+            col = accType + 'Imputed'
+            cols += [col]
+      
+        idx = cols.index('sleepImputed') #gets index of sleep label
+        # collects the sleep column from data frame assumes sleep if it is the highest imputed value
+        y = (np.argmax(e[cols].values, axis=1) == idx).astype('int')*2-1
```

Comment:
could lines 434-443 be simply replaced with something the equivalent of the below?

y = (e['sleepImputed']).astype('int')*2-1

Reply:
This can't be replaced like this - the awkward phrasing is to get round the fact that sleepImputed can take values between 0 and 1, and in that case we want to count the time as sleep if sleepImputed is higher than any other imputed value.

Reply:
Ah yes, you're completely right! This code is very nice actually.